app/assets/javascripts/provider/admin/apiconfig/services/service_discovery.js

```diff
@@ -57,10 +57,15 @@ document.addEventListener("DOMContentLoaded", function(e) {
     opt.text = val;
     return selectElem.appendChild(opt);
   };
+  
+  function safe_fetch(url, init) {
```

Comment:
You could call the function `customFetch`. It seems the `init` arg is not needed, right? 
Also, you could send a second arg named with default value like  `options = CUSTOM_FETCH_OPTIONS` and define that const before like:
```
const CUSTOM_FETCH_OPTIONS = { credentials: "same-origin" } 
```
I know this will be refactor in the (hopefully near) future, but that way would prepare the fn to a possible minor refactor accepting more request options.

Reply:
I think the idea is to harcode `credentials: "same-origin"` in the fetch call. Adding this `options` arg would make the method vulnerable again.

Reply:
if we want to make it "closed" we just pass the options in the original fetch call.